blueflood-elasticsearch/src/main/java/com/rackspacecloud/blueflood/io/ElasticsearchRestHelper.java

```diff
@@ -0,0 +1,261 @@
+        return INSTANCE;
+    }
+
+    private ElasticsearchRestHelper(){
+        logger.info("Creating a new instance of ElasticsearchRestHelper...");
+        Configuration config = Configuration.getInstance();
+        this.baseUrl = String.format("http://%s",
+                config.getStringProperty(ElasticIOConfig.ELASTICSEARCH_HOST_FOR_REST_CLIENT));
+        this.closeableHttpClient = HttpClientBuilder.create().build();
+    }
+
+    public String fetchEvents(String tenantId, Map<String, List<String>> query) throws IOException {
+        //Example URL: localhost:9200/metric_metadata/metrics/_search";
+        String url = String.format("%s/%s/%s/_search?routing=%s", baseUrl,
+                EventElasticSearchIO.EVENT_INDEX, EventElasticSearchIO.ES_TYPE, tenantId);
+
+        HttpPost httpPost = new HttpPost(url);
+        httpPost.setHeaders(getHeaders());
+
+        String queryDslString = getDslString(query);
+        CloseableHttpResponse response = null;
+
+        try {
+            HttpEntity httpEntity = new NStringEntity(queryDslString, ContentType.APPLICATION_JSON);
+            httpPost.setEntity(httpEntity);
+
+            response = closeableHttpClient.execute(httpPost);
+            String str = EntityUtils.toString(response.getEntity());
+            EntityUtils.consume(response.getEntity());
+            return str;
+        }
+        catch(Exception e){
+            if((e instanceof HttpResponseException) && (response != null))
+                logger.error("Status code: " + response.getStatusLine().getStatusCode() + " with msg: " + e.getMessage());
+
+            throw new RuntimeException(e.getMessage(), e);
+        }
+        finally {
+            response.close();
+        }
+    }
+
+    private String getDslString(Map<String, List<String>> query) {
+        String tagsValue = extractFieldFromQuery(Event.FieldLabels.tags.toString(), query);
+        String untilValue = extractFieldFromQuery(Event.untilParameterName, query);
+        String fromValue = extractFieldFromQuery(Event.fromParameterName, query);
+
+        String tagsQString = "";
+
+        if (StringUtils.isNotEmpty(tagsValue))
+            tagsQString = String.format("{\"term\":{\"%s\":\"%s\"}}", Event.FieldLabels.tags.toString(), tagsValue);
+
+        String rangeQueryString;
+
+        if (StringUtils.isNotEmpty(untilValue) && StringUtils.isNotEmpty(fromValue)) {
+            rangeQueryString = String.format("{\"range\":{\"when\":{\"from\":%d,\"to\":%d}}}",
+                    Long.parseLong(fromValue), Long.parseLong(untilValue));
+        } else if (StringUtils.isNotEmpty(untilValue)) {
+            rangeQueryString = String.format("{\"range\":{\"when\":{\"to\":%d}}}", Long.parseLong(untilValue));
+        } else if (StringUtils.isNotEmpty(fromValue)) {
+            rangeQueryString = String.format("{\"range\":{\"when\":{\"from\":%d}}}", Long.parseLong(fromValue));
+        } else {
+            // TODO: LOG THIS INVALID CASE
```

Comment:
Log it now so the TODO doesn't get forgotten about.

Reply:
Done.. :-)